agent/agentcontainers/api.go

```diff
@@ -583,10 +584,10 @@ func (api *API) processUpdatedContainersLocked(ctx context.Context, updated code
 		if dc.Container != nil {
 			if !api.devcontainerNames[dc.Name] {
 				// If the devcontainer name wasn't set via terraform, we
-				// use the containers friendly name as a fallback which
-				// will keep changing as the devcontainer is recreated.
-				// TODO(mafredri): Parse the container label (i.e. devcontainer.json) for customization.
-				dc.Name = safeFriendlyName(dc.Container.FriendlyName)
+				// will attempt to create an agent name based on the workspace
+				// folder's name. If that is not possible, we will fall back
+				// to using the container's friendly name.
+				dc.Name = safeAgentName(filepath.Base(dc.WorkspaceFolder), dc.Container.FriendlyName)
```

Comment:
If the agent is running on Windows, but controlling linux devcontainers, we'll need to use `path.Base` here. 
```suggestion
				dc.Name = safeAgentName(path.Base(dc.WorkspaceFolder), dc.Container.FriendlyName)
```

Reply:
Better to be safe than sorry so I've done `path.Base(filepath.ToSlash(...))`